src/com.spoiledcat.git.ui/Editor/UI/HierarchyWindowInterface.cs

```diff
@@ -0,0 +1,64 @@
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Unity.VersionControl.Git
+{
+    [InitializeOnLoad]
+    public class HierarchyWindowInterface
+    {
+        static HierarchyWindowInterface()
+        {
+            EditorApplication.hierarchyWindowItemOnGUI += OnHierarchyItemTryToDrawStatusIcon;
```

Comment:
This shouldn't be subscribed at `InitializeOnLoad` time, because there is no guarantee that this callback will happen after all of the git bits have been initialized by `ApplicationManagerBase`. You probably haven't noticed any issues because you already have git set up in your project, so in most cases this callback happens at a safe time, but at times when git needs to be set up or there's an update happening, this callback will get called before the setup is done, and the `ProjectWindowInterface.GetStatusIconForAssetGUID(guid)` call below will likely throw an exception.

To make sure there's no race conditions at startup, this should be done in a `Initialize` method like https://github.com/spoiledcat/git-for-unity/blob/main/src/com.spoiledcat.git.ui/Editor/UI/ProjectWindowInterface.cs#L39, and the call to it should be added in https://github.com/spoiledcat/git-for-unity/blob/main/src/com.spoiledcat.git.ui/Editor/ApplicationManager.cs#L36C35-L36C35

Reply:
You're right, your way is much better integrated with the tool ;) I have done as you asked, thanks!